src/ExpressReceiver.ts

```diff
@@ -64,25 +68,52 @@ export default class ExpressReceiver implements Receiver {
     // tslint:disable-next-line: align
     }, 3001);
 
-    const event: ReceiverEvent = {
-      body: req.body,
-      ack: async (response: any): Promise<void> => {
-        if (isAcknowledged) {
-          throw new ReceiverMultipleAckError();
-        }
-        isAcknowledged = true;
-        if (!response) {
-          res.send('');
-        } else if (typeof response === 'string') {
-          res.send(response);
-        } else {
-          res.json(response);
-        }
-      },
-    };
+    let event: ReceiverEvent;
+    let storedResponse = undefined;
+    if (this.processBeforeResponse) {
```

Comment:
i think there's a lot of overlap/reptition between the true branch and false branch of this `if`. i think we could reduce that repetition by putting the `if` inside the `ack` function implementation.